src/index.js

```diff
@@ -72,6 +70,39 @@ class Hue {
         const color = getRandomColor()
         this.setColors(ids, color)
     }
+
+    updateLight = async (id, state) => {
+        const lightState = await this.readLight(id)
+        if(JSON.stringify(lightState) === JSON.stringify(state)){
+            return
+        }
```

Comment:
Had some weird stuff if it updates to the exact same state, because technically you are resetting the light state to nothing initially. Added this check so that if you are calling `updateLight` with the same state it already is then it should not change.